infer/environment.cc

```diff
@@ -1003,9 +1003,9 @@ Environment::processBinding(core::Context ctx, const cfg::CFG &inWhat, cfg::Bind
                             const optional<cfg::BasicBlock::BlockExitCondInfo> &parentUpdateKnowledgeReceiver) {
     try {
         core::TypeAndOrigins tp;
-        bool noLoopChecking = cfg::isa_instruction<cfg::Alias>(bind.value) ||
-                              cfg::isa_instruction<cfg::LoadArg>(bind.value) ||
-                              bind.bind.variable == cfg::LocalRef::selfVariable();
+        bool noLoopChecking =
+            cfg::isa_instruction<cfg::Alias>(bind.value) || cfg::isa_instruction<cfg::LoadArg>(bind.value) ||
+            bind.bind.variable == cfg::LocalRef::selfVariable() || bindMinLoops == cfg::CFG::MIN_LOOP_NEVER_READ;
```

Comment:
Conceptually, these variables which are never read could be thought of as having their types "pinned" to `T.anything`.

I tried that in an initial version but I couldn't figure out all the places where `pinnedType[var]` is populated and it led to some spurious errors.